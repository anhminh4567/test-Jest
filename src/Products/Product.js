import React from "react";

const Product = ({items = []}) =>{
    if(items.length <= 0){
        return <div>No item found!</div> 
    }else{
        return <div>{items.map( (item,index) => {
            //return <h4 key={index} >item</h4>
            return <ol>
            
                <li >{item}</li>
            </ol>
        } )}</div>
    }
}
export default Product




