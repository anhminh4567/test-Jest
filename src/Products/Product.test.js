import React from "react";
import renderer from 'react-test-renderer';

import Products from './Product'

test("render when there are no item", () =>{ 
    const tree =renderer.create(<Products></Products>).toJSON();
    expect(tree).toMatchSnapshot();
    }
)
test("render when there are one item", () => {
    const tree =renderer.create(<Products items={["anything"]}></Products>).toJSON();
    expect(tree).toMatchSnapshot();

    }
)
test("render when there are Many item", () => {
    const tree =renderer.create(<Products items={["anything","asdfasd","just print stuff"]}></Products>).toJSON();
    expect(tree).toMatchSnapshot();

    }
)




