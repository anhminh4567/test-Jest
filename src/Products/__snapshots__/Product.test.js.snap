// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render when there are Many item 1`] = `
<div>
  <ol>
    <li>
      anything
    </li>
  </ol>
  <ol>
    <li>
      asdfasd
    </li>
  </ol>
  <ol>
    <li>
      just print stuff
    </li>
  </ol>
</div>
`;

exports[`render when there are no item 1`] = `
<div>
  No item found!
</div>
`;

exports[`render when there are one item 1`] = `
<div>
  <ol>
    <li>
      anything
    </li>
  </ol>
</div>
`;
